system_tests/bigquery.py

```diff
@@ -0,0 +1,224 @@
+        dataset = CLIENT.dataset(DATASET_NAME)
+        self.assertFalse(dataset.exists())
+        dataset.create()
+        self.to_delete.append(dataset)
+        tables_to_create = [
+            'new%d' % (1000 * time.time(),),
+            'newer%d' % (1000 * time.time(),),
+            'newest%d' % (1000 * time.time(),),
+        ]
+        full_name = bigquery.SchemaField('full_name', 'STRING',
+                                         mode='REQUIRED')
+        age = bigquery.SchemaField('age', 'INTEGER', mode='REQUIRED')
+        for table_name in tables_to_create:
+            table = dataset.table(table_name, schema=[full_name, age])
+            table.create()
+            self.to_delete.insert(0, table)
+
+        # Retrieve the tables.
+        all_tables, _ = dataset.list_tables()
+        created = [table for table in all_tables
+                   if table.name in tables_to_create and
+                   table._dataset.name == DATASET_NAME]
+        self.assertEqual(len(created), len(tables_to_create))
+
+    def test_patch_table(self):
+        DATASET_NAME = 'system_tests'
+        dataset = CLIENT.dataset(DATASET_NAME)
+        self.assertFalse(dataset.exists())
+        dataset.create()
+        self.to_delete.append(dataset)
+        TABLE_NAME = 'test_table'
+        full_name = bigquery.SchemaField('full_name', 'STRING',
+                                         mode='REQUIRED')
+        age = bigquery.SchemaField('age', 'INTEGER', mode='REQUIRED')
+        table = dataset.table(TABLE_NAME, schema=[full_name, age])
+        self.assertFalse(table.exists())
+        table.create()
+        self.to_delete.insert(0, table)
+        self.assertTrue(table.exists())
+        self.assertEqual(table.friendly_name, None)
+        self.assertEqual(table.description, None)
+        table.patch(friendly_name='Friendly', description='Description')
+        self.assertEqual(table.friendly_name, 'Friendly')
+        self.assertEqual(table.description, 'Description')
+
+    def test_update_table(self):
+        DATASET_NAME = 'system_tests'
+        dataset = CLIENT.dataset(DATASET_NAME)
+        self.assertFalse(dataset.exists())
+        dataset.create()
+        self.to_delete.append(dataset)
+        TABLE_NAME = 'test_table'
+        full_name = bigquery.SchemaField('full_name', 'STRING',
+                                         mode='REQUIRED')
+        age = bigquery.SchemaField('age', 'INTEGER', mode='REQUIRED')
+        table = dataset.table(TABLE_NAME, schema=[full_name, age])
+        self.assertFalse(table.exists())
+        table.create()
+        self.to_delete.insert(0, table)
+        self.assertTrue(table.exists())
+        voter = bigquery.SchemaField('voter', 'BOOLEAN', mode='NULLABLE')
+        schema = table.schema
+        schema.append(voter)
+        table.schema = schema
+        table.update()
+        self.assertEqual(len(table.schema), len(schema))
+        for found, expected in zip(table.schema, schema):
```

Comment:
The order is fixed (it maps onto the order in CSV and JSON (which doesn't have per-row column names mapped).